lisa/operating_system.py

```diff
@@ -385,14 +385,27 @@ def capture_system_information(self, saved_path: Path) -> None:
         from lisa.tools import Chmod, Find
 
         find_tool = self._node.tools[Find]
-        file_list = find_tool.find_files(
-            self._node.get_pure_path("/var/log/azure/"),
-            file_type="f",
-            sudo=True,
-            ignore_not_exist=True,
-        )
+        try:
+            file_list = find_tool.find_files(
+                self._node.get_pure_path("/var/log/azure/"),
+                file_type="f",
+                sudo=True,
+                ignore_not_exist=True,
+            )
+        except Exception:
+            # On systems like docker sudo is not present, so try without sudo
+            file_list = find_tool.find_files(
```

Comment:
I thought it again. we shouldn't handle it one by one for docker in the node.py

1. Change the logic of support_sudo to check if sudo supported for docker.
2. Update execute_async, not to raise exception, if support_sudo is not here.

Reply:
Nice! This would be much cleaner